### ISSUE TITLE
Equal height causes form input to loose on-screen keyboard as you focus (Android phones)

### ISSUE THREAD

**Issue by johnpalaganas** [wet-boew/wet-boew#8070]
Hello,

**WET:** 4.0.23
**Device:** Android phone (Nougat)

Any form input inside the Equal Height causes the on-screen keyboard to disappear. You cannot type anything at all.

Specifics to `<input type="text">` and `<textarea>`.

Whats going on?

**Sample code**
```
<div class="wb-eqht">
    <div class="col-md-5" style="background-color: #ccc">
        <h2>Column A</h2>
        <p> Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque justo dolor, condimentum ac placerat vel, sodales a nibh. In ac nisi vel arcu egestas mattis. Nulla mollis accumsan ex vitae vestibulum. Mauris quis urna euismod, blandit mauris id, sagittis dolor. Mauris sit amet dolor ornare, malesuada sapien a, malesuada lacus. Cras porta sapien id felis egestas pharetra. Cras ullamcorper odio ac mattis molestie. Vestibulum eget nulla eget quam aliquet tincidunt nec vel augue. Vestibulum sit amet rhoncus est. Morbi et libero egestas, mattis tellus quis, feugiat sapien. Proin at urna eget elit consequat blandit.</p>
        <form>
            <div class="form-group">
                <label for="email">Email</label>
                <input id="email" class="form-control" value="" type="text">
            </div>
            <div class="form-group">
                <label for="comments">Comments</label>
                <textarea id="comments" class="form-control" rows="6"></textarea>
            </div>
        </form>
    </div>
    <div class="col-md-5" style="background-color: #fff">
        <h2>Column B</h2>
        <p>Suspendisse metus orci, hendrerit ut sollicitudin a, porttitor ut ligula. Phasellus pellentesque ac tellus sed lobortis. Nam convallis mauris dolor, vitae placerat ligula euismod sed. Vestibulum vitae ipsum quis dui convallis consectetur sit amet id eros. Sed ultricies non quam ut ornare. Nunc sit amet consectetur orci, id porttitor urna. In at eros id diam fermentum varius. Maecenas elementum quis nulla nec eleifend. Curabitur tellus mauris, sollicitudin quis velit a, condimentum condimentum augue. Nunc lorem odio, tincidunt id elit ut, ullamcorper commodo nunc. In in consectetur arcu. Cras vitae nulla pharetra, dapibus lacus rutrum, hendrerit leo. Vivamus vel euismod risus, at rutrum arcu.</p>
    </div>
</div>
```

**Comment by shawnthompson** [comment]
I created this working example of your code to test.

https://shawnthompson.github.io/wet-boew-form-android/demos/index-en.html

Let me know if that's exactly what you have.

**Comment by shawnthompson** [comment]
Here's the same example not using wb-eqht

https://shawnthompson.github.io/wet-boew-form-android/demos/index-fr.html

I tested in browserstack and can confirm that the one using equal heights does stop the keyboard from popping up.

![image](https://user-images.githubusercontent.com/2575266/27924424-46048c5c-624f-11e7-9971-b516b8fd1c7b.png)

**Comment by johnpalaganas** [comment]
Hi Shawn,

Thanks! I personally don't have an Android phone but I sent it to someone who does. Will get back to you!

As a "fix" we just removed the (wb-eqht) Equal height plugin and just use a CSS solution instead to make both columns have the same height.

John

**Comment by johnpalaganas** [comment]
Got a confirmation.
https://shawnthompson.github.io/wet-boew-form-android/demos/index-fr.html
Works! The keyboard does not disappear.

https://shawnthompson.github.io/wet-boew-form-android/demos/index-en.html
Does not work, keyboard disappears.

Thanks,
John

**Comment by EricDunsworth** [comment]
I believe I know why that issue is so prominent in mobile browsers. Upon touching any text-based form input situated inside an equalized parent element, the mobile OS' on-screen keyboard will usually appear... which causes the browser's vertical height to diminish.  The equalization plugin probably interprets that as a resize event, then steals/resets focus as it attempts to re-equalize.

Since desktop browsers' mobile simulators don't usually simulate mobile OS' on-screen keyboards, the issue isn't as easy to reproduce in those. However, it can be reproduced if a user attempts to manually simulate what would've automatically happened in a mobile OS' browser when an input was touched.

Here's how to reproduce it in the desktop versions of Firefox and IE (doesn't work in Chrome/Opera since they normally lose focus when resizing browser windows):
1. Open a non-maximized Firefox or IE window.
2. Visit https://shawnthompson.github.io/wet-boew-form-android/demos/index-en.html.
3. Click the second input field (should cause a focus outline and text cursor to appear).
4. Vertically resize the browser window. The second input will lose focus.
5. Repeat steps 1-5 on https://shawnthompson.github.io/wet-boew-form-android/demos/index-fr.html. The second input will retain focus.

**Comment by johnpalaganas** [comment]
You're right Eric, the plugin interprets that as a resize event, then steals/resets focus as it attempts to re-equalize like you said. Thanks for the steps to reproduce it. Will not be using the plugin since there's unforeseen issues with it. Implementing a CSS solution seems to be more practical.

**Comment by EricDunsworth** [comment]
@johnpalaganas The plugin could probably be enhanced to resolve it, but I'm not in a position to fix it atm.

In case you're interested, long ago [I started developing a CSS-only equalization solution for browsers that natively support flexbox](https://github.com/EricDunsworth/wet-boew/compare/v4.0-eqht-flexbox), but never got around to finishing it off. https://github.com/wet-boew/wet-boew/issues/6418#issuecomment-161438432 contains a more detailed overview of the state it was in when I last dealt with it.
